App.js

```diff
@@ -40,19 +44,43 @@ class App extends Component{
     this.fetchApiData();
   }
 
+  sortDataById(){
+    const newData = this.state.data.sort((a,b) => a.id - b.id);
```

Comment:
Sort mutate array - so it doesn't create new instance. (thats why probably you need extraData property)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

You should first copy an array to create new one and then sort and put it into state.